examples/scenes/src/download.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright 2022 the Vello Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use std::io::Seek;
+use std::path::{Path, PathBuf};
+
+use anyhow::{bail, Context, Result};
+use byte_unit::Byte;
+use clap::Args;
+use inquire::Confirm;
+use std::io::Read;
+mod default_downloads;
+
+#[derive(Args, Debug)]
+pub(crate) struct Download {
+    #[clap(long)]
+    /// Directory to download the files into
+    #[clap(default_value_os_t = default_directory())]
+    pub directory: PathBuf,
+    /// Set of files to download. Use `name@url` format to specify a file prefix
+    downloads: Option<Vec<String>>,
+    /// Whether to automatically install the default set of files
+    #[clap(long)]
+    auto: bool,
+    /// The size limit for each individual file (ignored if the default files are downloaded)
+    #[clap(long, default_value = "10 MB")]
+    size_limit: Byte,
+}
+
+fn default_directory() -> PathBuf {
+    let mut result = Path::new(env!("CARGO_MANIFEST_DIR"))
+        .parent()
+        .unwrap()
+        .join("assets");
+    result.push("downloads");
+    result
+}
+
+impl Download {
+    pub fn action(&self) -> Result<()> {
+        let mut to_download = vec![];
+        if let Some(downloads) = &self.downloads {
+            to_download = downloads
+                .iter()
+                .map(|it| Self::parse_download(it))
+                .collect();
+        } else {
+            let mut accepted = self.auto;
+            let downloads = default_downloads::default_downloads()
+                .into_iter()
+                .filter(|it| {
+                    let file = it.file_path(&self.directory);
+                    !file.exists()
+                })
+                .collect::<Vec<_>>();
+            if !accepted {
+                if !downloads.is_empty() {
+                    println!(
+                        "Would you like to download a set of default lottie files? These files are:"
+                    );
+                    for download in &downloads {
+                        let builtin = download.builtin.as_ref().unwrap();
+                        println!(
+                            "{} ({}) under license {} from {}",
+                            download.name,
+                            byte_unit::Byte::from_bytes(builtin.expected_size.into())
+                                .get_appropriate_unit(false),
+                            builtin.license,
+                            builtin.info
+                        );
+                    }
+
```

Comment:
Do we really *need* to do this? 

Personally I think the with_winit demo should be abstracted into something we can import here, and vello_svg - And keep that work TODO when we get around to that.

Reply:
The point of printing the sizes is in case downloads are going to take up too much disk space/network usage - if e.g. you're on a limited internet connection

Similarly, the point of printing the licenses is so that people know what they're agreeing to

Checking that 350 items have the sameish size (as there is surely some variation), then counting them and multiplying. Then manually checking that each license is the same is pointless busiwork.

My concern is not about how long the downloads will take

Reply:
I'll add it as a followup task in #14